### PR TITLE
Make menus and placeholders display appropriate custom key combos.

### LINKED ISSUE #4039: Show custom hotkeys all over the application
### Checklist

- [X] I'm reporting a problem with Chatterino
- [X] I've verified that I'm running **the most** recent nightly build or stable release
- [X] I've looked for my problem on the [wiki](https://wiki.chatterino.com/Help/)
- [X] I've searched the [issues and pull requests](https://github.com/Chatterino/chatterino2/issues?q=) for similar looking reports

### Describe your issue

Instead of showing the default hotkey, show the custom one if it exists.
Some lines I found, but probably there are more.

https://github.com/Chatterino/chatterino2/blob/4c2e97bea68f2c13031f5125f4c17f3e0142a8f9/src/widgets/settingspages/GeneralPage.cpp#L562-L563
https://github.com/Chatterino/chatterino2/blob/4c2e97bea68f2c13031f5125f4c17f3e0142a8f9/src/widgets/settingspages/GeneralPage.cpp#L196
https://github.com/Chatterino/chatterino2/blob/4c2e97bea68f2c13031f5125f4c17f3e0142a8f9/src/widgets/dialogs/SettingsDialog.cpp#L88
https://github.com/Chatterino/chatterino2/blob/4c2e97bea68f2c13031f5125f4c17f3e0142a8f9/src/widgets/Notebook.cpp#L380-L383

### Screenshots

N/A

### OS and Chatterino Version

Chatterino Nightly 2.3.5 (commit 4c2e97be) Windows 10

### PR DESCRIPTION
**Pull request by Mm2PL**
Pull request checklist:

- [x] `CHANGELOG.md` was updated, if applicable

# Description

Closes #4039.
Todo:
- [x] make `neededArguments` work
- [-] implement it everywhere:
  - [x] SplitHeader 

  - [-] Settings
    - [x] search placeholder
    - [x] Hide similar messages setting
    - [x] prefs button setting
  - [x] Notebook (hide tabs)
  - [ ] ~~SplitInput~~ that's a Qt menu.
  - [ ] something more?
- [ ] Consider showing multiple shortcuts when possible (?)
- [x] Update documentation

All the commits mentioning scuffed-ness need some kind of relayout or app restart to update the key bind description.

### CODE REVIEW

**Comment by l4ejyetrgf** [comment]
If your messing with hotkeys, your may fix #3474 as well

**Comment by jammehcow** [comment]
> If your messing with hotkeys, your may fix #3474 as well

That should be a different PR as it has nothing to do with the scope of this one.

**Comment by Mm2PL** [comment]
> If your messing with hotkeys, your may fix https://github.com/Chatterino/chatterino2/issues/3474 as well

Getting side-tracked and introducing out-of-scope changes is the best way to never finish a PR. It's better to finish one and then start another. Custom hotkeys was already an awfully large PR (scope-wise, not code) and I honestly don't want to do anything that big.

File: src/controllers/hotkeys/HotkeyController.hpp [hunk 1]
```diff
@@ -114,6 +121,13 @@ class HotkeyController final : public Singleton
      **/
     static void showHotkeyError(const std::shared_ptr<Hotkey> &hotkey,
                                 QString warning);
+    /**
+     * @brief finds a Hotkey matching category, action and neededArguments.
+     * Accepted if and only if the category matches, the action matches and neededArguments match(TODO)
+     * */
```
Comment (pajlada) [inline src/controllers/hotkeys/HotkeyController.hpp:127]: Is this still TODO?
If no, could you update with the logic for how `neededArguments` should be handled? i.e. if there's no direct match, should it look for non-direct matches?

**Review by pajlada — CHANGES_REQUESTED** [review]
just comment nitpick, works as expected

some hotkeys don't update on change while others do but that's ok with me as a first effort, especially since this is already an improvement of usability

File: src/controllers/hotkeys/HotkeyController.hpp (hunk 1 above)
Comment (Mm2PL) [reply src/controllers/hotkeys/HotkeyController.hpp:127]: I forgot to update this, it should work. I use `neededArguments` a bunch, for example src/widgets/splits/SplitHeader.cpp:353, it requires hotkey arguments like {"split"}.

**Review by pajlada — APPROVED** [review]

**Comment by pajlada** [comment]
@Mm2PL Feel free to merge in when you're happy!

**Comment by pajlada** [comment]
Skipping FreeBSD build on Cirrus for now

**Pull request merged**